### PR TITLE
Fix pins, add VIA support for Stream Cheap 2x4

### PR DESCRIPTION
**Pull request by mpfl**
## Description

Added VIA support for the Stream Cheap 2x4.

While doing this, I noticed that the existing pin definition for the Stream Cheap 2x4 seemed entirely arbitrary. I have fixed the pins so that they are in a logical order.

I did not touch the Stream Cheap 2x3 and 2x5 configurations as I do not have them to test on.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

Nil

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/stream_cheap/2x4/rules.mk [hunk 1]
```diff
@@ -20,3 +20,4 @@ BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
 BLUETOOTH_ENABLE = no       # Enable Bluetooth
 AUDIO_ENABLE = no           # Audio output
+VIA_ENABLE = yes			# VIA support
```
Comment (fauxpark) [inline keyboards/handwired/stream_cheap/2x4/rules.mk:23]:
```suggestion
```
This goes in a separate rules.mk under the `via` keymap.

File: keyboards/handwired/stream_cheap/2x4/config.h [hunk 2]
```diff
@@ -3,19 +3,19 @@
...
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
+#define VENDOR_ID       0x7363 // Stream Cheap
 #define PRODUCT_ID      0x1214
 #define DEVICE_VER      0x0001
-#define MANUFACTURER    Kyle Hart
+#define MANUFACTURER    Stream Cheap
 #define PRODUCT         Stream Cheap 2x4
+#define DESCRIPTION     Stream Cheap 2x4
```
Comment (fauxpark) [inline keyboards/handwired/stream_cheap/2x4/config.h:11]:
```suggestion
#define PRODUCT         2x4
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/stream_cheap/2x4/rules.mk (hunk 1 above)
Comment (mpfl) [reply keyboards/handwired/stream_cheap/2x4/rules.mk]: Addressed in #4d5e4f1

File: keyboards/handwired/stream_cheap/2x4/config.h (hunk 2 above)
Comment (mpfl) [reply keyboards/handwired/stream_cheap/2x4/config.h]: Addressed in 4d5e4f1

File: keyboards/handwired/stream_cheap/2x4/keymaps/via/rules.mk [hunk 3]
```diff
@@ -0,0 +1,10 @@
+VIA_ENABLE = yes
+# LTO: link time optimization makes the build take slightly longer
+# but makes the resulting .hex file smaller, which allows you to
+# fit more features into smaller MCUs:
+LTO_ENABLE = yes
+# Support for these features make the hex file larger, but we want 'em:
+MOUSEKEY_ENABLE = yes     # Allow mapping of mouse control keys
+EXTRAKEY_ENABLE = yes     # Allow audio & system control keys
+COMMAND_ENABLE = yes      # Commands for debug and configuration
+NKRO_ENABLE = yes         # Support USB N-key roll over.
```
Comment (fauxpark) [inline keyboards/handwired/stream_cheap/2x4/keymaps/via/rules.mk:10]:
```suggestion
VIA_ENABLE = yes
LTO_ENABLE = yes
```
Command is not all that useful on a board of this size. The rest are already enabled at keyboard level.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/stream_cheap/2x4/keymaps/via/rules.mk (hunk 3 above)
Comment (mpfl) [reply keyboards/handwired/stream_cheap/2x4/keymaps/via/rules.mk]: Addressed in 3bb3a04

**Review by fauxpark — APPROVED** [review]

File: keyboards/handwired/stream_cheap/2x4/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/handwired/stream_cheap/2x4/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/stream_cheap/2x4/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+
+enum layers {
+  NORMAL_LAYER = 0,
+  SECOND_LAYER = 1,
+  THIRD_LAYER = 2,
+  FOURTH_LAYER = 3
```
Comment (drashna) [inline keyboards/handwired/stream_cheap/2x4/keymaps/via/keymap.c:8]: No need to specify the numbers.  Enums start at 0 and automatically increment up.
```suggestion
    NORMAL_LAYER,
    SECOND_LAYER,
    THIRD_LAYER,
    FOURTH_LAYER,
```

File: keyboards/handwired/stream_cheap/2x4/2x4.h [hunk 6]
```diff
@@ -3,8 +3,8 @@
 #include "quantum.h"
 
 #define LAYOUT_ortho_2x4(\
-	K00, K01, K02,K03, \
-	K04, K05, K06,K07  \
+	K00, K01, K02, K03, \
+	K04, K05, K06, K07  \
```
Comment (drashna) [inline keyboards/handwired/stream_cheap/2x4/2x4.h:7]:
```suggestion
    K00, K01, K02, K03, \
    K04, K05, K06, K07  \
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/stream_cheap/2x4/keymaps/via/keymap.c (hunk 4 above)
Comment (mpfl) [reply keyboards/handwired/stream_cheap/2x4/keymaps/via/keymap.c]: Addressed in 9be12f6

File: keyboards/handwired/stream_cheap/2x4/keymaps/via/keymap.c (hunk 5 above)
Comment (mpfl) [reply keyboards/handwired/stream_cheap/2x4/keymaps/via/keymap.c]: Addressed in 9be12f6

**Review by drashna — APPROVED** [review]

**Comment by luigi311** [comment]
Also looks like the defaults you provide for the via profile are useless without the use of VIA.

**Pull request merged**